paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/procedure/RollbackProcedure.java

```diff
@@ -61,15 +64,34 @@ public StructType outputType() {
     @Override
     public InternalRow[] call(InternalRow args) {
         Identifier tableIdent = toIdentifier(args.getString(0), PARAMETERS[0].name());
-        String version = args.getString(1);
+        String version = args.isNullAt(1) ? null : args.getString(1);
+        Long snapshot = args.isNullAt(2) ? null : args.getLong(2);
+        String tag = args.isNullAt(3) ? null : args.getString(3);
+        if ((version != null && snapshot != null)
+                || (snapshot != null && tag != null)
+                || (tag != null && version != null)) {
+            throw new IllegalArgumentException(
+                    "only can set one of version/snapshot/tag in RollbackProcedure.");
+        }
+
+        if (version == null && snapshot == null && tag == null) {
+            throw new IllegalArgumentException(
+                    "must set one of version/snapshot/tag in RollbackProcedure.");
+        }
 
         return modifyPaimonTable(
                 tableIdent,
                 table -> {
-                    if (version.chars().allMatch(Character::isDigit)) {
-                        table.rollbackTo(Long.parseLong(version));
+                    if (snapshot != null) {
+                        table.rollbackTo(snapshot);
+                    } else if (tag != null) {
+                        table.rollbackTo(tag);
                     } else {
-                        table.rollbackTo(version);
+                        if (version.chars().allMatch(Character::isDigit)) {
+                            table.rollbackTo(Long.parseLong(version));
+                        } else {
+                            table.rollbackTo(version);
+                        }
```

Comment:
Based on the above modifications.   snapshot != null ? table.rollbackTo(snapshot) : table.rollbackTo(tag);